From 51f9fba287ff7c8b10e0d5291cd66e78d40b6b8f Mon Sep 17 00:00:00 2001
From: Gal Pressman <galp@mellanox.com>
Date: Sun, 19 Mar 2017 14:10:41 +0200
Subject: [PATCH 4/7] ethtool: Fix SFF 8079 cable technology bit parsing

According to the transceiver compliance code definition in the spec, bits
2 & 3 in the 8th byte are indication of active/passive cable, and not
specifically related to FC/copper.

Fixes: 2edf56749abe ("ethtool: Addition of -m option to dump module eeprom")
Signed-off-by: Gal Pressman <galp@mellanox.com>
Signed-off-by: John W. Linville <linville@tuxdriver.com>
(cherry picked from commit e33c8841f26090031d571fddd71dab06f56ab1bf)
---
 sfpid.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sfpid.c b/sfpid.c
index fd6415c..1732e5e 100644
--- a/sfpid.c
+++ b/sfpid.c
@@ -137,9 +137,9 @@ static void sff8079_show_transceiver(const __u8 *id)
 	if (id[8] & (1 << 4))
 		printf("%s FC: Longwave laser (LL)\n", pfx);
 	if (id[8] & (1 << 3))
-		printf("%s FC: Copper Active\n", pfx);
+		printf("%s Active Cable\n", pfx);
 	if (id[8] & (1 << 2))
-		printf("%s FC: Copper Passive\n", pfx);
+		printf("%s Passive Cable\n", pfx);
 	if (id[8] & (1 << 1))
 		printf("%s FC: Copper FC-BaseT\n", pfx);
 	/* Fibre Channel transmission media */
-- 
1.8.3.1

